eth/core/chain_writer.go

```diff
@@ -92,8 +92,8 @@ func (bc *blockchain) Prepare(ctx context.Context, number uint64) {
 		Nonce:      types.BlockNonce{},
 	}
 
-	// We update the base fee in the txpool to the next base fee.
-	bc.tp.SetBaseFee(header.BaseFee)
+	// We update the txpool with the new block information.
+	bc.tp.Prepare(header.Number, header.Time, header.BaseFee)
```

Comment:
tbh I think we can remove all this Prepare() stuff and use channels, I lowkey think more efficient too.